### ISSUE TITLE
Volumes mounted by TrueCrypt are visible/accessible to other users

### ISSUE THREAD

**Issue by herzbube** [CipherShed/CipherShed#22]
On Mac OS X, when I mount a TrueCrypt volume from a file container while logged in as user A, I can then switch to another user B and view the mounted volume's content (e.g. in the Finder, or in a Terminal.app session). I believe this is a bug, as the content of the TrueCrypt volume should remain private to the user who mounted it. I don't know enough about the underlying issues to lay the blame on any one in particular (Mac OS X, TrueCrypt, FUSE?), but what I definitely can say is that I cannot trust my Mac to be left alone while a TrueCrypt volume is still mounted.

This is how my mounted volumes' mount points look like inside a Terminal.app session. As you can see, the TrueCrypt volume `PRIVATE` is mounted with permissions that make it wide open for any user to snoop around inside.

```
caradhras:~ --> ls -l /Volumes/
total 72
lrwxr-xr-x   1 root   admin      1 27 Jul 17:52 Macintosh HD -> /
drwxrwxrwx@  1 admin  staff  16384 31 Dez  1979 PRIVATE
drwx------   1 admin  staff  16384  9 Jun 17:06 data
```

What I would like to see here is that the `PRIVATE` volume has the same set of permissions as the `data` volume (which is a Samba network share mounted by Mac OS X). Ideally, the Finder should not display the `PRIVATE` volume at all, but if it does, the user should not be able to access the volume's content.

I am using TrueCrypt 7.1a on Mac OS X 10.9.4. The issue is not new, though, it already existed in previous versions of TrueCrypt; I don't recall the specific version I used then, but I submitted a problem report on the original TrueCrypt website in September 2009 - without getting any response.

I am aware that CipherShed has not made its initial rebranding release yet, so please forgive me for reporting this already. However, because CipherShed is based on the TrueCrypt 7.1a code base, I am quite confident that the issue will be present in your initial release as well.

**Comment by Deleetdk** [comment]
This is also a problem on Windows 8.1, using the last safe version of TrueCrypt. Devices mounted on user A are available to all other users.

**Comment by GigabyteProductions** [comment]
I think that's a Windows issue, though. I don't think Windows has a permissions system for drives themselves. You'd have to use a file system with a permission system, such as NTFS, and basically implement a whitelist with the ACL options.

**Comment by Deleetdk** [comment]
I am using NTFS formatted drives.

**Comment by GigabyteProductions** [comment]
Can other users access the volume when you have the permissions setup to only let you see?

**Comment by rockihack** [comment]
> Devices mounted on user A are available to all other users.

As far as I know all *admins* can access all mounted drives.
Did you try with restricted users?

**Comment by Deleetdk** [comment]
> Did you try with restricted users?

Good idea. I made a new user without administrator rights but it can still see the mounted drives.

**Comment by chromeronin** [comment]
Still an issue with MacOS 10.15.5 Catalina and VeraCrypt 1.24-Update4.
When I mount a volume as one user, another logged on user on the Mac can also see the contents of the mounted veracrypt volume. The other user is NOT an administrator user.
